filament/backend/src/vulkan/caching/VulkanPipelineLayoutCache.h

```diff
@@ -35,13 +35,27 @@ class VulkanPipelineLayoutCache {
 
     void terminate() noexcept;
 
-    using PipelineLayoutKey = std::array<VkDescriptorSetLayout,
-            VulkanDescriptorSetLayout::UNIQUE_DESCRIPTOR_SET_COUNT>;
+    struct PushConstantKey {
+        uint8_t stage;// We have one set of push constant per shader stage (fragment, vertex, etc).
+        uint8_t size;
+        // Note that there is also an offset parameter for push constants, but
+        // we always assume our update range will have the offset 0.
+    };
+
+    struct PipelineLayoutKey {
+        using DescriptorSetLayoutArray = std::array<VkDescriptorSetLayout,
+                VulkanDescriptorSetLayout::UNIQUE_DESCRIPTOR_SET_COUNT>;
+        DescriptorSetLayoutArray descSetLayouts = {};
+        PushConstantKey pushConstant[Program::SHADER_TYPE_COUNT] = {};
+    };
```

Comment:
Maybe assert the size of `PipelineLayoutKey` is what we expect to ensure no padding bytes are inserted by the compiler.

Reply:
done